### PR TITLE
Feature: add removeFiles and  onRemoveFiles

### PR DESCRIPTION
**Pull request by Morteza-Jenabzadeh**
**What kind of change does this PR introduce?**
- [ ] bugfix
- [x] feature
- [ ] refactoring / style
- [ ] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [x] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**

- [x] Yes, I've updated the documentation
- [ ] Not relevant

**Summary**
This feature helps developers to remove files from either `acceptedFiles` or `fileRejections`. 
This PR has already been implemented by @estrattonbailey in the closed PR #942 the problem was using `find` which is not supported in ie :))
so I used `indexof` that is well supported in all browsers.

**Does this PR introduce a breaking change?**
Shouldn't

**Other information**
If you check the code there is a double iteration 
the first one is to remove files from the state:
```javascript
const acceptedFiles =  state.acceptedFiles && state.acceptedFiles.
    filter(acceptedFile => files.indexOf(acceptedFile) ===-1);
    const fileRejections = state.fileRejections && state.fileRejections.
    filter(rejectedFile => files.indexOf(rejectedFile) ===-1);
```
and the second one is to guarantee that `onRemovedFiles` callback return just removedfiles.   
```javascript
 removedFiles.push(...files.filter(x => state.acceptedFiles.indexOf(x)!==-1));
 removedFiles.push(...files.filter(x => state.fileRejections.indexOf(x)!==-1));
``` 
I also couldn't find a way to run linter to format staged files (husky should handle but it didn't), and there is no prettier config file
please give me a hint about it.

I hope people like it :)

### CODE REVIEW

**Comment by rolandjitsu** [comment]
@Morteza-Jenabzadeh the main reason we didn't want this sort of feature is because it adds complexity.

Because the file processing is async, you need to ensure the `removeFiles` func call doesn't occur while that is happening, but to queue it after. So managing the timing of callbacks becomes more complicated.

I really think this sort of logic should be in the user's apps.

**Comment by Morteza-Jenabzadeh** [comment]
well, I hve seen many ppl asked for this feature #805  and you agreed with this feature https://github.com/react-dropzone/react-dropzone/issues/805#issuecomment-580034406 and thought it would be nice to handle this request :)

**Review by ALexanderMarginal — APPROVED** [review]

**Comment by rolandjitsu** [comment]
@Morteza-Jenabzadeh ok, let me review it this weekend.

**Review by rolandjitsu — CHANGES_REQUESTED** [review]

File: examples/removeFiles/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+You can use `removeFiles` function to remove any file either in acceptedFiles ro fileRejections.
```
[the remark this answers is not in the document]
Comment (Morteza-Jenabzadeh) [reply examples/removeFiles/README.md:1]: ok, sounds reasonable.

**Comment by uguraktas** [comment]
Has this feature been released? I couldn't find anything in the documentation and code. @morijenab @rolandjitsu

**Comment by BartusZak** [comment]
The same here :(

```ts
export type DropzoneOptions = Pick<React.HTMLProps<HTMLElement>, PropTypes> & {
  accept?: string | string[];
  minSize?: number;
  maxSize?: number;
  maxFiles?: number;
  preventDropOnDocument?: boolean;
  noClick?: boolean;
  noKeyboard?: boolean;
  noDrag?: boolean;
  noDragEventsBubbling?: boolean;
  disabled?: boolean;
  onDrop?: <T extends File>(acceptedFiles: T[], fileRejections: FileRejection[], event: DropEvent) => void;
  onDropAccepted?: <T extends File>(files: T[], event: DropEvent) => void;
  onDropRejected?: (fileRejections: FileRejection[], event: DropEvent) => void;
  getFilesFromEvent?: (event: DropEvent) => Promise<Array<File | DataTransferItem>>;
  onFileDialogCancel?: () => void;
  validator?: <T extends File>(file: T) => FileError | FileError[] | null;
};
``

**Pull request closed without merging**